CleanWeather/Scenes/CityList/CityListInteractor.swift

```diff
@@ -46,7 +46,10 @@ extension CityListInteractorImpl: CityListInteractor {
         }
     }
     
-    func didSelectCityCell(id: String) {
-        router.navigateToCityDetails(id: id)
+    func didSelectCityCell(with id: String) {
+        let filter = cityWeather.filter { $0.id == id }
+        if let weather = filter.first {
```

Comment:
if let weather = cityWeather.first( { $0.id == id }  {

Reply:
guard let is better

Reply:
Zrobiłem, ale nie rozumiem dlaczego guard let w tym wypadku lepsze.

Reply:
To pytanie chyba umknęło